### PR TITLE
Luhn's Algorithm in CPP

### LINKED ISSUE #320: Luhn's Algorithm
## 🚀 Feature

Luhn's Algorithm in CPP language

## Assignment
[C++]
Hello maintainers!  
As a participant/CA for GSSOC '21,  
I wish to contribute & resolve this issue with proper indentation, comments & sample I/O in C++.
Kindly assign me these issues under GSSOC '21...

## Pitch
This simple algorithm validates credit cards.

### PR DESCRIPTION
**Pull request by Suvraneel**
### Description

Luhn's Algorithm for credit card validation in CPP

### Checklist

- [x] I've edited the `README.md` and link to my code.
- [x] This is in response to an issue #320 assigned to me under GSSoC'21

## Related Issues

resolves #320

@shreyanspoddar

### CODE REVIEW

**Comment by shreyanspoddar** [comment]
@OjusWiZard Please review it.

File: Code/C++/Luhns_algorithm.cpp [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+}
+
+//main driver code
+int main()
+{
+    //declare character array
+    char card[16];
+    cout << "Enter credit card number:" << endl;
+    //reads a line from input and stores it into the 'card' character array
+    gets (card);
```
Comment (OjusWiZard) [inline Code/C++/Luhns_algorithm.cpp:63]: Don't use gets()

File: Code/C++/Luhns_algorithm.cpp [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+Std O/P 2:
+Enter the credit card number:
+5987654024681353
+ + 1 + 9 + 7 + 7 + 3 + 5 + 8 + 0 + 4 + 4 + 3 + 8 + 2 + 3 + 1 + 3         = 68  (Invalid)
+
+Std O/P 3:
+Enter the credit card number:
+59876540246813550
+ + 1 + 9 + 7 + 7 + 3 + 5 + 8 + 0 + 4 + 4 + 3 + 8 + 2 + 3 + 1 + 5 + 0     = 70  (Invalid)
+[Invalid because card number must have exactly 16 digits.]
```
Comment (OjusWiZard) [inline Code/C++/Luhns_algorithm.cpp:93]: Write the output just as it appears on the console.

**Review by OjusWiZard — CHANGES_REQUESTED** [review]

File: Code/C++/Luhns_algorithm.cpp (hunk 2 above)
Comment (Suvraneel) [reply Code/C++/Luhns_algorithm.cpp:93]: Thats how it appears in the console...

File: Code/C++/Luhns_algorithm.cpp (hunk 1 above)
Comment (Suvraneel) [reply Code/C++/Luhns_algorithm.cpp]: Resolved

**Comment by Suvraneel** [comment]
@OjusWiZard  Done :)

**Comment by Suvraneel** [comment]
@rudrakshi99  @tarun26091999  @OjusWiZard  :)

File: Code/C++/Luhns_algorithm.cpp (hunk 2 above)
Comment (OjusWiZard) [reply Code/C++/Luhns_algorithm.cpp:96]: `+` is never printed by the code in between the numbers. Yet, it still appears in your I/O exampls.

File: Code/C++/Luhns_algorithm.cpp [hunk 3]
```diff
@@ -0,0 +1,97 @@
...
+    cout << "\t" << ((int)strlen(card) == 16 && checksum%10==0 ? "(Valid)":"(Invalid)") << endl;       
+    //the card number must have exactly 16 digits   
+    if ((int)strlen(card) != 16)
+        cout << "[Invalid because card number must have exactly 16 digits.]" << endl;            
+    return 0; 
+}
+
+/*
+Time Complexity = O(1)
+Space Complexity = O(1)
```
Comment (OjusWiZard) [inline Code/C++/Luhns_algorithm.cpp:80]: Shift complexities below the test cases.

**Review by OjusWiZard — CHANGES_REQUESTED** [review]

**Comment by Suvraneel** [comment]
@OjusWiZard Fixed the code...  & resolved requested changes

File: Code/C++/Luhns_algorithm.cpp (hunk 3 above)
Comment (Suvraneel) [reply Code/C++/Luhns_algorithm.cpp]: My bad... It now does print the '+'

**Comment by plazzy99** [comment]
Resolve conflicts

**Review by OjusWiZard — APPROVED** [review]

**Comment by krishna-NIT** [comment]
@Suvraneel  Level tag is missed, @OjusWiZard please add it

**Pull request merged**